ddtrace/profiling/collector/pytorch.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import absolute_import
+
+import abc
+import logging
+import typing
+
+import attr
+
+from ddtrace._trace.tracer import Tracer
+from ddtrace.internal.datadog.profiling import ddup
+from ddtrace.profiling import _threading
+from ddtrace.profiling import collector
+from ddtrace.profiling.recorder import Recorder
+from ddtrace.settings.profiling import config
+from ddtrace.vendor import wrapt
+
+
+LOG = logging.getLogger(__name__)
+
+pytorch_events_limit = config.pytorch.events_limit or 1_000_000
+
+
+class _WrappedTorchProfiler(wrapt.ObjectProxy):
+    def __init__(
+        self,
+        wrapped: typing.Any,
+        recorder: Recorder,
+        tracer: typing.Optional[Tracer],
+    ) -> None:
+        wrapt.ObjectProxy.__init__(self, wrapped)
+        self.on_trace_ready = handle_torch_trace
+        self._self_recorder = recorder
+        self._self_tracer = tracer
+
+
+@attr.s
+class MLProfilerCollector(collector.CaptureSamplerCollector):
+    """Record ML framework (i.e. pytorch) profiler usage."""
+
+    tracer = attr.ib(default=None)
+    _original = attr.ib(init=False, repr=False, type=typing.Any, cmp=False)
+
+    @abc.abstractmethod
+    def _get_original(self):
+        # type: (...) -> typing.Any
+        pass
+
+    @abc.abstractmethod
+    def _set_original(
+        self,
+        value,  # type: typing.Any
+    ):
+        # type: (...) -> None
+        pass
+
+    def _start_service(self):
+        # type: (...) -> None
+        """Start collecting framework profiler usage."""
+        try:
+            import torch
+        except ImportError as e:
+            raise collector.CollectorUnavailable(e)
+        self._torch_module = torch
+        self.patch()
+        super(MLProfilerCollector, self)._start_service()
+
+    def _stop_service(self):
+        # type: (...) -> None
+        """Stop collecting framework profiler usage."""
+        super(MLProfilerCollector, self)._stop_service()
+        self.unpatch()
+
+    def patch(self):
+        # type: (...) -> None
```

Comment:
```suggestion
    def patch(self) -> None :
```

Reply:
What's the issue with the type annotation here?

Reply:
i.e. should I apply this everywhere in this file?

Reply:
https://github.com/DataDog/dd-trace-py/issues/11729 to track